Test TestLocalExecutor.test_execution(unlimited) sometimes fail

The test sometimes fails:

```python
=================================== FAILURES ===================================
_________________ TestLocalExecutor.test_execution[unlimited] __________________
tests/executors/test_local_executor.py:114: in test_execution
    self._test_execute(parallelism=parallelism)
/usr/local/lib/python3.11/unittest/mock.py:1378: in patched
    return func(*newargs, **newkeywargs)
tests/executors/test_local_executor.py:91: in _test_execute
    executor.end()
airflow/executors/local_executor.py:230: in end
    proc.join()
/usr/local/lib/python3.11/multiprocessing/process.py:149: in join
    res = self._popen.wait(timeout)
/usr/local/lib/python3.11/multiprocessing/popen_fork.py:43: in wait
    return self.poll(os.WNOHANG if timeout == 0.0 else 0)
/usr/local/lib/python3.11/multiprocessing/popen_fork.py:27: in poll
    pid, sts = os.waitpid(self.pid, flag)
E   Failed: Timeout >60.0s
----------------------------- Captured stdout call -----------------------------
[2025-01-01T13:45:44.503+0000] {local_executor.py:60} INFO - Worker starting up pid=288
[2025-01-01T13:45:44.519+0000] {local_executor.py:60} INFO - Worker starting up pid=290
[2025-01-01T13:45:44.523+0000] {local_executor.py:215} INFO - Shutting down LocalExecutor; waiting for running tasks to finish.  Signal again if you don't want to wait.
[2025-01-01T13:45:44.525+0000] {local_executor.py:96} ERROR - uhoh
Traceback (most recent call last):
  File "/opt/airflow/airflow/executors/local_executor.py", line 92, in _run_worker
    _execute_work(log, workload)
  File "/opt/airflow/airflow/executors/local_executor.py", line 113, in _execute_work
    supervise(
  File "/usr/local/lib/python3.11/unittest/mock.py", line 1124, in __call__
    return self._mock_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/unittest/mock.py", line 1128, in _mock_call
    return self._execute_mock_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/unittest/mock.py", line 1189, in _execute_mock_call
    result = effect(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/tests/executors/test_local_executor.py", line 71, in fake_supervise
    raise RuntimeError("fake failure")
RuntimeError: fake failure
[2025-01-01T13:45:44.527+0000] {local_executor.py:60} INFO - Worker starting up pid=305
[2025-01-01T13:45:44.564+0000] {local_executor.py:60} INFO - Worker starting up pid=318
[2025-01-01T13:45:44.580+0000] {local_executor.py:60} INFO - Worker starting up pid=297
------------------------------ Captured log call -------------------------------
INFO     airflow.executors.local_executor.LocalExecutor:local_executor.py:215 Shutting down LocalExecutor; waiting for running tasks to finish.  Signal again if you don't want to wait.
----------- generated xml file: /files/test_result-core-postgres.xml -----------
```

For example https://github.com/apache/airflow/actions/runs/12571140824/job/35041860269#step:6:2934

--- a/airflow-core/tests/unit/executors/test_local_executor.py
+++ b/airflow-core/tests/unit/executors/test_local_executor.py
@@ -78,7 +78,6 @@ class TestLocalExecutor:
 
         executor.end()
 
-    @skip_spawn_mp_start
     @mock.patch("airflow.sdk.execution_time.supervisor.supervise")
     def test_execution(self, mock_supervise):
         success_tis = [
@@ -101,8 +100,12 @@ class TestLocalExecutor:
         fail_ti = success_tis[0].model_copy(update={"id": uuid7(), "task_id": "failure"})
 
         # We just mock both styles here, only one will be hit though
+        has_failed_once = False
+
         def fake_supervise(ti, **kwargs):
-            if ti.id == fail_ti.id:
+            nonlocal has_failed_once
+            if ti.id == fail_ti.id and not has_failed_once:
+                has_failed_once = True
                 raise RuntimeError("fake failure")
             return 0
 

